### PR TITLE
MULE-10754 Remove VoidMuleEvent

### PR DESCRIPTION
**Pull request by dfeist**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/routing/SynchronousUntilSuccessfulProcessingStrategyTestCase.java [hunk 1]
```diff
@@ -158,14 +159,14 @@ public void successfulExecutionWithAckExpression() throws Exception {
...
     when(mockRoute.process(any(Event.class))).thenReturn(null);
     SynchronousUntilSuccessfulProcessingStrategy processingStrategy = createProcessingStrategy();
-    assertThat(processingStrategy.route(testEvent(), getTestFlow(muleContext)), is(testEvent()));
+    assertThat(processingStrategy.route(testEvent(), getTestFlow(muleContext)), is(nullValue()));
   }
 
   @Test
   public void successfulWithNullEventResponseFromRoute() throws Exception {
-    when(mockRoute.process(any(Event.class))).thenReturn(VoidMuleEvent.getInstance());
+    when(mockRoute.process(any(Event.class))).thenReturn(null);
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/routing/SynchronousUntilSuccessfulProcessingStrategyTestCase.java]: is this null returning valid now?

File: core/src/main/java/org/mule/runtime/core/routing/EventGroup.java [hunk 2]
```diff
@@ -329,11 +328,11 @@ public Event getMessageCollectionEvent() {
             Event.builder(lastEvent).message(builder.build()).session(getMergedSession(muleEvents)).build();
         return muleEvent;
       } else {
-        return VoidMuleEvent.getInstance();
+        return null;//VoidMuleEvent.getInstance();
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/routing/EventGroup.java]: remove comments

File: core/src/main/java/org/mule/runtime/core/routing/SynchronousUntilSuccessfulProcessingStrategy.java [hunk 3]
```diff
@@ -40,17 +39,15 @@ protected Event doRoute(Event event, FlowConstruct flow) throws MuleException {
       for (int i = 0; i <= getUntilSuccessfulConfiguration().getMaxRetries(); i++) {
         try {
           Event successEvent = processResponseThroughAckResponseExpression(processEvent(retryEvent));
+          if (successEvent == null) {
+            return null;
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/routing/SynchronousUntilSuccessfulProcessingStrategy.java]: is returning null ok here?

File: tests/integration/src/test/java/org/mule/test/integration/routing/outbound/MulticasterMixedSyncAsyncTestCase.java [hunk 4]
```diff
@@ -37,7 +37,7 @@ public void testMixedMulticast() throws Exception {
     List<Apple> results =
         ((List<InternalMessage>) result.getPayload().getValue()).stream().map(msg -> (Apple) msg.getPayload().getValue())
             .collect(toList());
-    assertEquals(2, results.size());
+    assertEquals(3, results.size());
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/integration/routing/outbound/MulticasterMixedSyncAsyncTestCase.java]: what's the reason of this change?

Comment (dfeist) [inline tests/integration/src/test/java/org/mule/test/integration/routing/outbound/MulticasterMixedSyncAsyncTestCase.java]: This is a change in behaviour in MulticastRouter.  This is deliberate, and MulticastRouter will be going away in 4.0 anyway.

File: core/src/main/java/org/mule/runtime/core/routing/SynchronousUntilSuccessfulProcessingStrategy.java (hunk 3 above)
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/routing/SynchronousUntilSuccessfulProcessingStrategy.java]: Yes, nulls need to be propagated always.

File: core/src/main/java/org/mule/runtime/core/routing/EventGroup.java (hunk 2 above)
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/routing/EventGroup.java]: I commented out this to look at it later, but then found all tests passed anyway.  I need to review this.
